packages/solid-form/src/createField.tsx

```diff
@@ -0,0 +1,132 @@
+import { FieldApi } from '@tanstack/form-core'
+import {
+  createComponent,
+  createComputed,
+  createEffect,
+  createMemo,
+  createSignal,
+  onCleanup,
+  onMount,
+} from 'solid-js'
+import { formContext, useFormContext } from './formContext'
+
+import type { DeepKeys, DeepValue } from '@tanstack/form-core'
+import type { JSXElement } from 'solid-js'
+import type { CreateFieldOptions } from './types'
+
+declare module '@tanstack/form-core' {
+  // eslint-disable-next-line no-shadow
+  interface FieldApi<
+    TParentData,
+    TName extends DeepKeys<TParentData>,
+    TData = DeepValue<TParentData, TName>,
+  > {
+    Field: FieldComponent<TParentData>
+  }
+}
+
+export type CreateField<TParentData> = typeof createField<TParentData>
+
+// ugly way to trick solid into triggering updates for changes on the fieldApi
+function makeFieldReactive<FieldApiT extends FieldApi<any, any>>(
+  fieldApi: FieldApiT,
+): () => FieldApiT {
+  const [flag, setFlag] = createSignal(false)
+  const fieldApiMemo = createMemo(() => [flag(), fieldApi] as const)
+  const unsubscribeStore = fieldApi.store.subscribe(() => setFlag((f) => !f))
+  onCleanup(unsubscribeStore)
+  return () => fieldApiMemo()[1]
+}
```

Comment:
Let's see if we can't replace this with our `@tanstack/store` implementation from @wobsoriano (speaking of, if you @wobsoriano could review this PR as another Solid pro, that would be rad):

https://github.com/TanStack/store/pull/25